### PR TITLE
17599 - Pupil name and dob

### PR DESCRIPTION
**Pull request by pris54**
Rules to display DOB and middle names on all pupil lists.

### CODE REVIEW

File: admin/services/pupils-not-taking-check.service.js [hunk 1]
```diff
@@ -1,30 +1,45 @@
...
+    } else if (sortDirection === 'asc') {
+      return a.reason < b.reason ? -1 : 1
+    } else {
+      return a.reason < b.reason ? 1 : -1
+    }
+  })
+  return sortedPupilsList
+}
+
+pupilsNotTakingCheckService.pupils = async (schoolId, sortField, sortDirection) => {
```
Comment (GuyHarwood) [inline admin/services/pupils-not-taking-check.service.js]: this method name give me no idea what it does

File: admin/services/pupils-not-taking-check.service.js [hunk 2]
```diff
@@ -1,30 +1,45 @@
...
+  const pupils = await pupilDataService.sqlFindSortedPupilsWithAttendanceReasons(schoolId, sortField, sortDirection)
+  pupilIdentificationFlag.addIdentificationFlags(pupils)
+  return pupils
+}
+
+pupilsNotTakingCheckService.pupilsWithReasons = async (schoolId) => {
```
Comment (GuyHarwood) [inline admin/services/pupils-not-taking-check.service.js]: this method name give me no idea what it does

File: admin/services/pupils-not-taking-check.service.js (hunk 1 above)
Comment (pris54) [reply admin/services/pupils-not-taking-check.service.js]: @GuyHarwood I struggled with naming these two methods (see your other comment), I'm relying on the service name to know what it is about. Would `getPupils` be good enough?

File: admin/services/pupils-not-taking-check.service.js (hunk 2 above)
Comment (pris54) [reply admin/services/pupils-not-taking-check.service.js]: @GuyHarwood please see my previous reply. In this case would `getPupilsWithReasons` be enough?

File: admin/services/pupils-not-taking-check.service.js (hunk 1 above)
Comment (GuyHarwood) [reply admin/services/pupils-not-taking-check.service.js]: build the name through its purpose, and prefix it with get.

what is it doing exactly, sorting?

File: admin/services/pupils-not-taking-check.service.js (hunk 2 above)
Comment (GuyHarwood) [reply admin/services/pupils-not-taking-check.service.js]: it needs to be prefixed with get, if thats what its doing?

**Comment by pris54** [comment]
@GuyHarwood Please let me know if the new method names make more sense.

**Pull request merged**